### PR TITLE
fix(FEC-8717): when player ui is small the captions language options dropdown is cut

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

When the player is small, the drop-down menus are not fully shown (might be cut).
The change is to have the full overlay menus (like in mobile, using the browsers native dropdowns) - so all the options will be shown.

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/shell/shell.js [hunk 1]
```diff
@@ -301,7 +305,7 @@ class Shell extends BaseComponent {
 
     if (this.props.prePlayback) playerClasses.push(style.prePlayback);
     if (this.props.isCasting) playerClasses.push(`${__CSS_MODULE_PREFIX__}-casting`);
-    if (this.props.isMobile) playerClasses.push(style.touch);
+    if (this.props.isMobile || this.props.isSmallView) playerClasses.push(style.touch);
```
[the remark this answers is not in the document]
Comment (odedhutzler) [reply src/components/shell/shell.js]: Good point. It reduces the functionality of the player (only mute/unmute). no play button on the control bar.

lets talk about it

File: src/styles/style.css [hunk 2]
```diff
@@ -620,6 +620,43 @@ video::cue {
...
+  margin-right: 16px;
+}
+.player.size-sm:not(.touch) .smart-container-item.select-menu-item .dropdown,
+.player.size-sm:not(.touch) .smart-container-item.select-menu-item select {
+  float: right;
+}
+.player.size-sm:not(.touch) .smart-container-item.select-menu-item select {
+  text-align-last: right;
+}
+
```
Comment (odedhutzler) [inline src/styles/style.css]: move this to saas structure

File: src/components/engine-connector/engine-connector.js [hunk 3]
```diff
@@ -70,6 +71,13 @@ class EngineConnector extends BaseComponent {
       }
     });
 
+    this.eventManager.listen(this.player, this.player.Event.RESIZE, () => {
+      const el = document.getElementById(this.player.config.targetId);
+      if (el) {
+        this.props.updateIsSmallView(el.getClientRects()[0].width < PLAYER_SIZE_OPTIONS.SMALL);
```
Comment (yairans) [inline src/components/engine-connector/engine-connector.js]: use `getCBoundedRects`

File: src/components/overlay-action/overlay-action.js [hunk 4]
```diff
@@ -18,7 +18,8 @@ const mapStateToProps = state => ({
   isPlayingAdOrPlayback: isPlayingAdOrPlayback(state.engine),
   iconType: state.overlayAction.iconType,
   playerHover: state.shell.playerHover,
-  isMobile: state.shell.isMobile
+  isMobile: state.shell.isMobile,
+  isSmallView: state.shell.isSmallView
```
Comment (yairans) [inline src/components/overlay-action/overlay-action.js]: unused

File: src/components/engine-connector/engine-connector.js [hunk 5]
```diff
@@ -70,6 +71,13 @@ class EngineConnector extends BaseComponent {
       }
     });
 
+    this.eventManager.listen(this.player, this.player.Event.RESIZE, () => {
```
Comment (yairans) [inline src/components/engine-connector/engine-connector.js]: check if you can remove it and call to `updateIsSmallView` in `shell render`

File: src/reducers/shell.js [hunk 6]
```diff
@@ -84,6 +91,7 @@ export const actions = {
   addPlayerClass: (className: string) => ({type: types.ADD_PLAYER_CLASS, className}),
   removePlayerClass: (className: string) => ({type: types.REMOVE_PLAYER_CLASS, className}),
   updateIsMobile: (isMobile: boolean) => ({type: types.UPDATE_IS_MOBILE, isMobile}),
+  updateIsSmallView: (isSmallView: boolean) => ({type: types.UPDATE_IS_SMALL_VIEW, isSmallView}),
```
Comment (yairans) [inline src/reducers/shell.js]: change to `updateViewSize' which update the current size large/medium/small

File: src/styles/style.css [hunk 7]
```diff
@@ -620,6 +620,7 @@ video::cue {
 .player:not(.touch) .smart-container .smart-container-item.select-menu-item select {
   text-align-last: right;
 }
+
```
Comment (yairans) [inline src/styles/style.css]: delete

File: src/components/smart-container/_smart-container.scss [hunk 8]
```diff
@@ -95,39 +95,41 @@
...
-    width: 300px;
-    max-width: 100%;
-    margin: 16px auto;
-    color: rgba(244, 244, 244, 0.8);
-    white-space: nowrap;
-    text-align: left;
-    display: flex;
-    justify-content: space-between;
+.player {
+  &.touch, &.size-sm:not(.touch) {
```
Comment (yairans) [inline src/components/smart-container/_smart-container.scss]:
```suggestion
  &.touch, &.size-sm {
```

File: src/reducers/shell.js [hunk 9]
```diff
@@ -84,6 +91,7 @@ export const actions = {
   addPlayerClass: (className: string) => ({type: types.ADD_PLAYER_CLASS, className}),
   removePlayerClass: (className: string) => ({type: types.REMOVE_PLAYER_CLASS, className}),
   updateIsMobile: (isMobile: boolean) => ({type: types.UPDATE_IS_MOBILE, isMobile}),
+  updatePlayerSize: (playerSize: string) => ({type: types.UPDATE_PLAYER_SIZE, playerSize}),
```
Comment (yairans) [inline src/reducers/shell.js]:
```suggestion
  updatePlayerSize: (playerSize: number) => ({type: types.UPDATE_PLAYER_SIZE, playerSize}),
```

File: src/components/shell/shell.js [hunk 10]
```diff
@@ -41,7 +42,13 @@ const mapStateToProps = state => ({
  * @type {number}
  * @const
  */
-export const CONTROL_BAR_HOVER_DEFAULT_TIMEOUT: number = 3000;
+const CONTROL_BAR_HOVER_DEFAULT_TIMEOUT: number = 3000;
+
+const PLAYER_SIZE: {[size: string]: number} = {
+  SMALL: 480,
+  MEDIUM: 768,
+  LARGE: 1024
```
Comment (yairans) [inline src/components/shell/shell.js]: add `X-LARGE`

Comment (yairans) [reply src/components/shell/shell.js]: change to strings - "small" etc.

File: src/reducers/shell.js (hunk 9 above)
Comment (odedhutzler) [reply src/reducers/shell.js]: leaving as is, as the playerSize will be string

File: src/components/shell/shell.js [hunk 11]
```diff
@@ -313,9 +327,19 @@ class Shell extends BaseComponent {
...
+      playerClasses.push(style.sizeSm);
+      this.props.updatePlayerSize(PLAYER_SIZE.SMALL);
+    } else if (this.props.playerClientRect && this.props.playerClientRect.width <= PLAYER_BREAK_POINTS.MEDIUM) {
+      playerClasses.push(style.sizeMd);
+      this.props.updatePlayerSize(PLAYER_SIZE.MEDIUM);
+    } else if (this.props.playerClientRect && this.props.playerClientRect.width <= PLAYER_BREAK_POINTS.LARGE) {
+      playerClasses.push(style.sizeLg);
+      this.props.updatePlayerSize(PLAYER_SIZE.LARGE);
+    } else {
+      playerClasses.push(style.sizeLg);
```
Comment (yairans) [inline src/components/shell/shell.js]: remove. this breaks the default style. 
or add `sizeXlg` (and add a black style, otherwise it won't be added)

Comment (odedhutzler) [reply src/components/shell/shell.js]: removed

**Pull request merged**